WordPress/Classes/Models/Blog.m

```diff
@@ -439,6 +444,46 @@ - (BOOL)supports:(BlogFeature)feature
     }
 }
 
+- (BOOL)supportsPublicize
+{
+    // wpcom blog and publicize has not been disabled
+    if (self.isHostedAtWPcom && ![[self getOptionValue:OptionsKeyPublicizeDisabled] boolValue]) {
+        return YES;
+    }
```

Comment:
I think if `isHostedAtWPcom` is true but publicize is disabled, it should return `NO` immediately.